src/main/java/seedu/address/logic/commands/EditEventCommand.java

```diff
@@ -0,0 +1,240 @@
+package seedu.address.logic.commands;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_ADD_FRIENDNAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DATETIME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_DESCRIPTION;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_NAME;
+import static seedu.address.logic.parser.CliSyntax.PREFIX_REMOVE_FRIENDNAME;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+import seedu.address.commons.core.Messages;
+import seedu.address.commons.core.index.Index;
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.event.DateTime;
+import seedu.address.model.event.Event;
+import seedu.address.model.person.Description;
+import seedu.address.model.person.Name;
+
+/**
+ * Edits the details of an existing event in the address book.
+ */
+public class EditEventCommand extends Command {
+
+    public static final String COMMAND_WORD = "editevent";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Edits the details of the person identified "
+            + "by the index number used in the displayed person list. "
+            + "Existing values will be overwritten by the input values.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "[" + PREFIX_NAME + "NEW_EVENT_NAME] "
```

Comment:
I thought that since we will not be referencing any of the old values it is clear that if a name is provided, the old name will be replaced. Furthermore, this reduces the length of the command.

Reply:
Yea, I agree with Aryan on this one, I think it's more intuitive to recycle the same prefixes since we are now selecting by index.

Reply:
On another note, I had some issues with the command formatting in the UG, and I think the same problem exists here, so do get that fixed.